src/prefect/flows.py

```diff
@@ -1326,7 +1338,8 @@ async def deploy(
         cron: Optional[str] = None,
         rrule: Optional[str] = None,
         paused: Optional[bool] = None,
-        schedules: Optional[list[DeploymentScheduleCreate]] = None,
+        schedule: Optional[Schedule] = None,
+        schedules: Optional[list[Schedule]] = None,
```

Comment:
i think this still needs to accept a `DeploymentScheduleCreate` object for backwards compatibility - are you thinking type hints should always be future-facing though (which would make sense to me as a light-touch deprecation)?

Reply:
Yeah, my thought was that updating the type hints would be a good way to nudge users towards the new class, but the functionality would remain the same if a `DeploymentScheduleCreate` is provided. It shouldn't be too bad to maintain the current code path.